iroha/src/kura.rs

```diff
@@ -180,126 +189,252 @@ impl Default for Mode {
-    use std::path::Path;
+    use std::{num::NonZeroU64, path::Path};
 
     use eyre::{eyre, Result};
     use iroha_config::derive::Configurable;
     use serde::{Deserialize, Serialize};
 
     use super::Mode;
 
-    const DEFAULT_KURA_BLOCK_STORE_PATH: &str = "./blocks";
+    /// Default number of blocks per storage file
+    pub const DEFAULT_BLOCKS_PER_STORAGE_FILE: u64 = 1000_u64;
+    const DEFAULT_BLOCK_STORE_PATH: &str = "./blocks";
 
     /// Configuration of kura
     #[derive(Clone, Deserialize, Serialize, Debug, Configurable)]
     #[serde(rename_all = "UPPERCASE")]
+    #[config(env_prefix = "KURA_")]
     pub struct KuraConfiguration {
         /// Possible modes: `strict`, `fast`.
         #[serde(default)]
-        pub kura_init_mode: Mode,
+        pub init_mode: Mode,
         /// Path to the existing block store folder or path to create new folder.
-        #[serde(default = "default_kura_block_store_path")]
-        pub kura_block_store_path: String,
+        #[serde(default = "default_block_store_path")]
+        pub block_store_path: String,
+        /// Maximum number of blocks to write into single storage file
+        #[serde(default = "default_blocks_per_storage_file")]
+        pub blocks_per_storage_file: NonZeroU64,
     }
 
     impl Default for KuraConfiguration {
         fn default() -> Self {
             Self {
-                kura_init_mode: Mode::default(),
-                kura_block_store_path: default_kura_block_store_path(),
+                init_mode: Mode::default(),
+                block_store_path: default_block_store_path(),
+                blocks_per_storage_file: default_blocks_per_storage_file(),
             }
         }
     }
 
     impl KuraConfiguration {
-        /// Set `kura_block_store_path` configuration parameter - will overwrite the existing one.
+        /// Set `block_store_path` configuration parameter - will overwrite the existing one.
         ///
         /// # Errors
         /// If path is not valid this method will fail.
-        pub fn kura_block_store_path(&mut self, path: &Path) -> Result<()> {
-            self.kura_block_store_path = path
+        pub fn block_store_path(&mut self, path: &Path) -> Result<()> {
+            self.block_store_path = path
                 .to_str()
                 .ok_or_else(|| eyre!("Failed to yield slice from path"))?
                 .to_owned();
             Ok(())
         }
+
+        /// Set `blocks_per_storage_file` parameter - will owerwrite the existing one
+        ///
+        pub fn blocks_per_storage_file(&mut self, size: NonZeroU64) {
+            self.blocks_per_storage_file = size;
+        }
     }
 
-    fn default_kura_block_store_path() -> String {
-        DEFAULT_KURA_BLOCK_STORE_PATH.to_owned()
+    fn default_block_store_path() -> String {
+        DEFAULT_BLOCK_STORE_PATH.to_owned()
+    }
+
+    fn default_blocks_per_storage_file() -> NonZeroU64 {
+        #![allow(clippy::expect_used, clippy::restriction)]
```

Comment:
Ok, here's something I really want to consult about:
I think forcing clippy here is ugly, but simply can't find a better way. This error is not possible by design, the only way we could get it is by having typo-change in constant. So, is it ok to mute clippy for such a case?

Reply:
Thats ok here. You shouldn't allow lint groups (for example `clippy::restriction`) though

Generally using unwraps or expects is fine if you justify that they are unreachable.